### PR TITLE
replace, delete, update visitor

### PR DESCRIPTION
**Pull request by pawel-big-lebowski**
Signed-off-by: Pawel Leszczynski <leszczynski.pawel@gmail.com>

### Problem

Missing visitors for Spark v2 commands: `ReplaceTableAsSelect`, `DeleteFromTable`, `UpdateTable`, `MergeIntoTable`

Closes tasks from: #368

### Solution

Please describe your change as it relates to the problem, or bug fix, as well as any dependencies. If your change requires a schema change, please describe the schema modifcation(s) and whether it's a _backwards-incompatible_ or _backwards-compatible_ change, then select one of the following:

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [X] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [X] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [X] Your changes are accompanied by tests (_if relevant_)
- [X] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [ ] You've updated the [`CHANGELOG.md`](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)

### CODE REVIEW

File: integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/TableContentChangeVisitor.java [hunk 1]
```diff
@@ -30,30 +35,47 @@ public TableContentChangeVisitor(OpenLineageContext context) {
   public boolean isDefinedAt(LogicalPlan x) {
     return (x instanceof OverwriteByExpression)
         || (x instanceof OverwritePartitionsDynamic)
+        || (x instanceof DeleteFromTable)
+        || (x instanceof UpdateTable)
+        || (new IcebergHandler().hasClasses() && x instanceof ReplaceData)
```
[the remark this answers is not in the document]
Comment (mobuchowski) [reply integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/TableContentChangeVisitor.java:40]: nevermind - we cannot have static interface method 🙂

File: integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/CreateReplaceVisitor.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      tableProperties = ScalaConversionUtils.<String, String>fromMap(command.properties());
+      identifier = command.tableName();
+      schema = command.tableSchema();
+      stateChange = OVERWRITE;
+    } else {
+      ReplaceTableAsSelect command = (ReplaceTableAsSelect) x;
+      tableCatalog = command.catalog();
+      tableProperties = ScalaConversionUtils.<String, String>fromMap(command.properties());
+      identifier = command.tableName();
+      schema = command.tableSchema();
```
[the remark this answers is not in the document]
Comment (pawel-big-lebowski) [reply integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/CreateReplaceVisitor.java:73]: Would love to but I think it's not possible. `CreateTableAsSelect`, `ReplaceTable` and `ReplaceTableAsSelect` are Scala case classes with the same fields but they do not implement common interface nor extend common class. The fields are just named the same way which makes it difficult to extract to separate&common method.

File: integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/TableContentChangeVisitor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pawel-big-lebowski) [reply integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/TableContentChangeVisitor.java:40]: Indeed, the constructor with no variable assigned looks a bit dirty. On the other hand static methods are not allowed in interfaces unless they contain implementation.

File: integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/CreateReplaceVisitor.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+  public CreateReplaceVisitor(OpenLineageContext context) {
+    super(context);
+  }
+
+  @Override
+  public boolean isDefinedAt(LogicalPlan x) {
+    return (x instanceof CreateTableAsSelect)
+        || (x instanceof ReplaceTable)
+        || (x instanceof ReplaceTableAsSelect);
```
[the remark this answers is not in the document]
Comment (pawel-big-lebowski) [reply integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/CreateReplaceVisitor.java]: I also included integration test for that.

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**